File: classify-rest-webservice/src/main/java/com/remswork/project/alice/repo/UserRepository.java
package com.remswork.project.alice.repo;

import com.remswork.project.alice.model.UserDetail;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface UserRepository extends JpaRepository<UserDetail, Long> {

//    @Query("from UserDetail where username=:username LIMIT 1")
//    UserDetail getByUsername(@Param("username") String username);
}

File: classify-rest-webservice/src/main/java/io/classify/data/repository/UserRepository.kt
package io.classify.data.repository

import io.classify.data.entity.Student
import io.classify.data.entity.Teacher
import io.classify.data.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param

interface UserRepository : JpaRepository<User, Long> {

    @Query("from User where id=:userId")
    fun findById(@Param("userId") userId: Long): User

    @Query("from User where username=:username")
    fun findByUsername(@Param("username") username: String): User

    @Query("from Student as a join a.user as b where b.id=:userId")
    fun findStudent(@Param("userId") userId: Long): Student

    @Query("from Teacher as a join a.user as b where b.id=:userId")
    fun findTeacher(@Param("userId") userId: Long): Teacher
}